File: solution/f#/Nir-Kata/Parse.Dont.Validate/Sex.fs
namespace Nir_Kata.Parse.Dont.Validate.NIRDomain

module Sex =
    type Sex =
        private
        | M
        | F

    let parse (potentialSex: char) =
        match potentialSex with
        | '1' -> Some M
        | '2' -> Some F
        | _ -> None

    let value s = if (s = M) then 1 else 2

File: solution/c#/Nir-Kata/Parse.Dont.Validate/Sex.cs
using LanguageExt;

namespace Nir_Kata.Parse.Dont.Validate
{
    public enum Sex
    {
        M = 1,
        F = 2
    }

    public static class SexParser
    {
        public static Option<Sex> Parse(char potentialSex) =>
            potentialSex switch
            {
                '1' => Sex.M,
                '2' => Sex.F,
                _ => Option<Sex>.None
            };
    }
}